batch_job/src/batch_job_mesos.c

```diff
@@ -81,13 +81,28 @@ static void start_mesos_scheduler(struct batch_queue *q)
 	
 	    char *exe_py_path = string_format("%s/mf_mesos_scheduler", exe_dir_path);
 		char *ld_preload_str = NULL;
+		char *python_path = NULL;
 	
 		if(mesos_preload) {
 			ld_preload_str = string_format("LD_PRELOAD=%s", mesos_preload);
-		} 
-		char *envs[] = {ld_preload_str, NULL};
-	
-		char *mesos_python_path = xxstrdup(mesos_py_path);
+		}
```

Comment:
Is this fixing a bug in mesos only, or is this related to Kubernetes somehow?

Reply:
It is only related to Mesos. We used to setup Mesos on disc cluster, which requires users to specify the 
share library to use Mesos. So I made `ld_preload` a mandatory flag. When I tried to setup Mesos on AWS, I can expose all Mesos libs to the `PATH`, which does not require me to pass any args to `ld_preload`. So I think it is more reasonable to make it an Optional flag